lib/uri/http.rb

```diff
@@ -80,6 +80,42 @@ def request_uri
       url = @query ? "#@path?#@query" : @path.dup
       url.start_with?(?/.freeze) ? url : ?/ + url
     end
+
+    #
+    # == Description
+    #
+    # Returns the authority for an HTTP uri, as defined in
+    # https://datatracker.ietf.org/doc/html/rfc3986/#section-3.2.
+    #
+    #
+    # Example:
+    #
+    #     URI::HTTP.build(host: 'www.example.com', path: '/foo/bar').authority #=> "www.example.com"
+    #     URI::HTTP.build(host: 'www.example.com', port: 8000, path: '/foo/bar').authority #=> "www.example.com:8000"
+    #     URI::HTTP.build(host: 'www.example.com', port: 80, path: '/foo/bar').authority #=> "www.example.com"
+    #
+    def authority
+      port_string = port == default_port ? nil : ":#{port}"
```

Comment:
I would personally like to see a parentheses around this sub expression?

Reply:
Using a top level if statement would allow you to avoid string allocation on the most common case.